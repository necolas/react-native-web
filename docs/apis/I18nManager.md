# I18nManager

Control and set the layout and writing direction of the application. You must
set `dir="rtl"` (and should set `lang="${lang}"`) on the root element of your
app.

## Properties

**isRTL**: bool = false

Whether the application is currently in RTL mode.

## Methods

static **allowRTL**(allowRTL: bool)

Allow the application to display in RTL mode.

static **forceRTL**(allowRTL: bool)

Force the application to display in RTL mode.

static **setRTL**(allowRTL: bool)

Set the application to display in RTL mode. You will need to determine the
user's preferred locale and if it is an RTL language. (This is best done on the
server as it is notoriously inaccurate to deduce client-side.)
