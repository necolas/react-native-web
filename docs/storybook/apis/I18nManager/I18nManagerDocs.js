/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import RTLToggleExample from './examples/RTLToggle';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import UIExplorer, { DocItem } from '../../ui-explorer';

const sections = [
  {
    title: 'Properties',
    entries: [
      <DocItem
        name="isRTL"
        typeInfo="boolean = false"
        description="Whether the application is currently in RTL mode."
      />
    ]
  },
  {
    title: 'Methods',
    entries: [
      <DocItem
        name="static allowRTL"
        typeInfo="(allowRTL: boolean) => void"
        description="Allow the application to display in RTL mode."
      />,

      <DocItem
        name="static forceRTL"
        typeInfo="(forceRTL: boolean) => void"
        description="Force the application to display in RTL mode."
      />,

      <DocItem
        label="web"
        name="static setPreferredLanguageRTL"
        typeInfo="(isRTL: boolean) => void"
        description="Set the application's preferred writing direction to RTL. You will need to determine the user's preferred locale server-side (from HTTP headers) and decide whether it's an RTL language."
      />
    ]
  },
  {
    title: 'Examples',
    entries: [
      <DocItem
        description="Toggling LTR/RTL layout at runtime"
        example={{
          render: () => <RTLToggleExample />
        }}
      />
    ]
  }
];

storiesOf('APIs', module).add('I18nManager', () =>
  <UIExplorer
    description="Control and set the layout and writing direction of the application."
    sections={sections}
    title="I18nManager"
    url="apis/I18nManager"
  />
);
