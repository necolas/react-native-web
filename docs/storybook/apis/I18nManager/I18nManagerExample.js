import { storiesOf } from '@kadira/storybook';
import UIExplorer from '../../UIExplorer';
import { I18nManager, StyleSheet, TouchableHighlight, Text, View } from 'react-native';
import React, { Component } from 'react';

class I18nManagerExample extends Component {
  componentWillUnmount() {
    I18nManager.setPreferredLanguageRTL(false);
  }

  render() {
    return (
      <View style={styles.container}>
        <Text accessibilityRole="heading" style={styles.welcome}>
          LTR/RTL layout example!
        </Text>
        <Text style={styles.text}>
          The writing direction of text is automatically determined by the browser, independent of
          the global writing direction of the app.
        </Text>
        <Text style={[styles.text, styles.rtlText]}>
          أحب اللغة العربية
        </Text>
        <Text style={[styles.text, styles.textAlign]}>
          textAlign toggles
        </Text>
        <View style={styles.horizontal}>
          <View style={[styles.box, { backgroundColor: 'lightblue' }]}>
            <Text>One</Text>
          </View>
          <View style={[styles.box]}>
            <Text>Two</Text>
          </View>
        </View>
        <TouchableHighlight
          onPress={this._handleToggle}
          style={styles.toggle}
          underlayColor="rgba(0,0,0,0.25)"
        >
          <Text>Toggle LTR/RTL</Text>
        </TouchableHighlight>
      </View>
    );
  }

  _handleToggle = () => {
    I18nManager.setPreferredLanguageRTL(!I18nManager.isRTL);
    this.forceUpdate();
  };
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
    padding: 10
  },
  welcome: {
    fontSize: 28,
    marginVertical: 10
  },
  text: {
    color: '#333333',
    fontSize: 18,
    marginBottom: 5
  },
  textAlign: {
    textAlign: 'left'
  },
  horizontal: {
    flexDirection: 'row',
    marginVertical: 10
  },
  box: {
    borderWidth: 1,
    flex: 1
  },
  toggle: {
    alignSelf: 'center',
    borderColor: 'black',
    borderStyle: 'solid',
    borderWidth: 1,
    marginTop: 10,
    padding: 10
  }
});

const examples = [
  {
    title: 'RTL toggle',
    render: () => <I18nManagerExample />
  }
];

storiesOf('APIs', module).add('I18nManager', () =>
  <UIExplorer examples={examples} title="I18nManager" />
);
