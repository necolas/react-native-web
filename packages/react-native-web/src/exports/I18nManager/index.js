/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

type I18nManagerStatus = {
  allowRTL: (allowRTL: boolean) => void,
  forceRTL: (forceRTL: boolean) => void,
  getConstants: () => Constants
};

type Constants = {
  isRTL: boolean
};

const I18nManager: I18nManagerStatus = {
  allowRTL() {
    return;
  },
  forceRTL() {
    return;
  },
  getConstants(): Constants {
    return { isRTL: false };
  }
};

export default I18nManager;
