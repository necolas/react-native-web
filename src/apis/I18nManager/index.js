/**
 * Copyright (c) 2016-present, Nicolas Gallagher.
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule I18nManager
 * @flow
 */

import ExecutionEnvironment from 'fbjs/lib/ExecutionEnvironment';

type I18nManagerStatus = {
  allowRTL: (allowRTL: boolean) => void,
  forceRTL: (forceRTL: boolean) => void,
  setPreferredLanguageRTL: (setRTL: boolean) => void,
  isRTL: boolean
};

let isPreferredLanguageRTL = false;
let isRTLAllowed = true;
let isRTLForced = false;

const isRTL = () => {
  if (isRTLForced) {
    return true;
  }
  return isRTLAllowed && isPreferredLanguageRTL;
};

const onChange = () => {
  if (ExecutionEnvironment.canUseDOM) {
    if (document.documentElement && document.documentElement.setAttribute) {
      document.documentElement.setAttribute('dir', isRTL() ? 'rtl' : 'ltr');
    }
  }
};

const I18nManager: I18nManagerStatus = {
  allowRTL(bool) {
    isRTLAllowed = bool;
    onChange();
  },
  forceRTL(bool) {
    isRTLForced = bool;
    onChange();
  },
  setPreferredLanguageRTL(bool) {
    isPreferredLanguageRTL = bool;
    onChange();
  },
  get isRTL() {
    return isRTL();
  }
};

export default I18nManager;
