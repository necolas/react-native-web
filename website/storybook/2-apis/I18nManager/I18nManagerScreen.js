/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import RTLToggle from './examples/RTLToggle';
import React from 'react';
import UIExplorer, { Description, DocItem, Section, storiesOf } from '../../ui-explorer';

const I18nManagerScreen = () => (
  <UIExplorer title="I18nManager" url="2-apis/I18nManager">
    <Description>
      Control and query the layout and writing direction of the application.
    </Description>
    <Section title="Properties">
      <DocItem
        name="isRTL"
        typeInfo="boolean = false"
        description="Whether the application is currently in RTL mode."
      />

      <DocItem
        name="doLeftAndRightSwapInRTL"
        typeInfo="boolean = true"
        description="Whether the application swaps left/right styles in RTL mode."
      />
    </Section>

    <Section title="Methods">
      <DocItem
        name="allowRTL"
        typeInfo="(allowRTL: boolean) => void"
        description="Allow the application to display in RTL mode."
      />

      <DocItem
        name="forceRTL"
        typeInfo="(forceRTL: boolean) => void"
        description="Force the application to display in RTL mode."
      />

      <DocItem
        name="swapLeftAndRightInRTL"
        typeInfo="(flipStyles: boolean) => void"
        description="Control whether the application swaps left/right styles in RTL mode. Applications relying on start/end styles may prefer to disable automatic BiDi-flipping of left/right styles."
      />

      <DocItem
        label="web"
        name="setPreferredLanguageRTL"
        typeInfo="(isRTL: boolean) => void"
        description="Set the application's preferred writing direction to RTL. You may need to infer the user's preferred locale on the server (from HTTP headers) and decide whether it's an RTL language."
      />
    </Section>

    <Section title="Examples">
      <DocItem
        description="Toggling LTR/RTL layout at runtime"
        example={{
          render: () => <RTLToggle />
        }}
      />
    </Section>
  </UIExplorer>
);

storiesOf('APIs', module).add('I18nManager', I18nManagerScreen);
