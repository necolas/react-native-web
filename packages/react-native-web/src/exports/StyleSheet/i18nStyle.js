/**
 * Copyright (c) Nicolas Gallagher.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import I18nManager from '../I18nManager';
import multiplyStyleLengthValue from '../../modules/multiplyStyleLengthValue';

const emptyObject = {};

const borderTopLeftRadius = 'borderTopLeftRadius';
const borderTopRightRadius = 'borderTopRightRadius';
const borderBottomLeftRadius = 'borderBottomLeftRadius';
const borderBottomRightRadius = 'borderBottomRightRadius';
const borderLeftColor = 'borderLeftColor';
const borderLeftStyle = 'borderLeftStyle';
const borderLeftWidth = 'borderLeftWidth';
const borderRightColor = 'borderRightColor';
const borderRightStyle = 'borderRightStyle';
const borderRightWidth = 'borderRightWidth';
const right = 'right';
const marginLeft = 'marginLeft';
const marginRight = 'marginRight';
const paddingLeft = 'paddingLeft';
const paddingRight = 'paddingRight';
const left = 'left';

// Map of LTR property names to their BiDi equivalent.
const PROPERTIES_FLIP = {
  borderTopLeftRadius: borderTopRightRadius,
  borderTopRightRadius: borderTopLeftRadius,
  borderBottomLeftRadius: borderBottomRightRadius,
  borderBottomRightRadius: borderBottomLeftRadius,
  borderLeftColor: borderRightColor,
  borderLeftStyle: borderRightStyle,
  borderLeftWidth: borderRightWidth,
  borderRightColor: borderLeftColor,
  borderRightStyle: borderLeftStyle,
  borderRightWidth: borderLeftWidth,
  left: right,
  marginLeft: marginRight,
  marginRight: marginLeft,
  paddingLeft: paddingRight,
  paddingRight: paddingLeft,
  right: left
};

// Map of I18N property names to their LTR equivalent.
const PROPERTIES_I18N = {
  borderTopStartRadius: borderTopLeftRadius,
  borderTopEndRadius: borderTopRightRadius,
  borderBottomStartRadius: borderBottomLeftRadius,
  borderBottomEndRadius: borderBottomRightRadius,
  borderStartColor: borderLeftColor,
  borderStartStyle: borderLeftStyle,
  borderStartWidth: borderLeftWidth,
  borderEndColor: borderRightColor,
  borderEndStyle: borderRightStyle,
  borderEndWidth: borderRightWidth,
  end: right,
  marginStart: marginLeft,
  marginEnd: marginRight,
  paddingStart: paddingLeft,
  paddingEnd: paddingRight,
  start: left
};

const PROPERTIES_VALUE = {
  clear: true,
  float: true,
  textAlign: true
};

// Invert the sign of a numeric-like value
const additiveInverse = (value: string | number) => multiplyStyleLengthValue(value, -1);

const i18nStyle = <T: {| [key: string]: any |}>(originalStyle: T): T => {
  const { doLeftAndRightSwapInRTL, isRTL } = I18nManager.getConstants();
  const style = originalStyle || emptyObject;
  const frozenProps = {};
  const nextStyle = {};

  for (const originalProp in style) {
    if (!Object.prototype.hasOwnProperty.call(style, originalProp)) {
      continue;
    }
    const originalValue = style[originalProp];
    let prop = originalProp;
    let value = originalValue;

    // BiDi flip properties
    if (PROPERTIES_I18N.hasOwnProperty(originalProp)) {
      // convert start/end
      const convertedProp = PROPERTIES_I18N[originalProp];
      prop = isRTL ? PROPERTIES_FLIP[convertedProp] : convertedProp;
    } else if (isRTL && doLeftAndRightSwapInRTL && (PROPERTIES_FLIP[originalProp]: any)) {
      prop = PROPERTIES_FLIP[originalProp];
    }

    // BiDi flip values
    if (PROPERTIES_VALUE.hasOwnProperty(originalProp)) {
      if (originalValue === 'start') {
        value = isRTL ? 'right' : 'left';
      } else if (originalValue === 'end') {
        value = isRTL ? 'left' : 'right';
      } else if (isRTL && doLeftAndRightSwapInRTL) {
        if (originalValue === 'left') {
          value = 'right';
        } else if (originalValue === 'right') {
          value = 'left';
        }
      }
    }

    // BiDi flip transitionProperty value
    if (prop === 'transitionProperty') {
      // BiDi flip properties
      if (PROPERTIES_I18N.hasOwnProperty(value)) {
        // convert start/end
        const convertedValue = PROPERTIES_I18N[originalValue];
        value = isRTL ? PROPERTIES_FLIP[convertedValue] : convertedValue;
      } else if (isRTL && doLeftAndRightSwapInRTL) {
        const flippedValue = PROPERTIES_FLIP[(originalValue: any)];
        if (flippedValue != null) {
          value = flippedValue;
        }
      }
    }

    // Create finalized style
    if (isRTL && prop === 'textShadowOffset') {
      const invertedValue = additiveInverse((value: any).width);
      (value: any).width = invertedValue;
      nextStyle[prop] = value;
    } else if (!frozenProps[prop]) {
      nextStyle[prop] = value;
    }

    if (PROPERTIES_I18N[originalProp]) {
      frozenProps[prop] = true;
    }
  }

  // $FlowIgnore
  return nextStyle;
};

export default i18nStyle;
