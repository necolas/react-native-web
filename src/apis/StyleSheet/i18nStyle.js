/**
 * Copyright (c) 2016-present, Nicolas Gallagher.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @noflow
 */

import I18nManager from '../I18nManager';
import multiplyStyleLengthValue from '../../modules/multiplyStyleLengthValue';

const emptyObject = {};

/**
 * Map of property names to their BiDi equivalent.
 */
const PROPERTIES_TO_SWAP = {
  borderTopLeftRadius: 'borderTopRightRadius',
  borderTopRightRadius: 'borderTopLeftRadius',
  borderBottomLeftRadius: 'borderBottomRightRadius',
  borderBottomRightRadius: 'borderBottomLeftRadius',
  borderLeftColor: 'borderRightColor',
  borderLeftStyle: 'borderRightStyle',
  borderLeftWidth: 'borderRightWidth',
  borderRightColor: 'borderLeftColor',
  borderRightWidth: 'borderLeftWidth',
  borderRightStyle: 'borderLeftStyle',
  left: 'right',
  marginLeft: 'marginRight',
  marginRight: 'marginLeft',
  paddingLeft: 'paddingRight',
  paddingRight: 'paddingLeft',
  right: 'left'
};

const PROPERTIES_SWAP_LEFT_RIGHT = {
  clear: true,
  float: true,
  textAlign: true
};

/**
 * Invert the sign of a numeric-like value
 */
const additiveInverse = (value: String | Number) => multiplyStyleLengthValue(value, -1);

/**
 * BiDi flip the given property.
 */
const flipProperty = (prop: String): String => {
  return PROPERTIES_TO_SWAP.hasOwnProperty(prop) ? PROPERTIES_TO_SWAP[prop] : prop;
};

const swapLeftRight = (value: String): String => {
  return value === 'left' ? 'right' : value === 'right' ? 'left' : value;
};

const i18nStyle = originalStyle => {
  if (!I18nManager.isRTL) {
    return originalStyle;
  }

  const style = originalStyle || emptyObject;
  const nextStyle = {};

  for (const prop in style) {
    if (!Object.prototype.hasOwnProperty.call(style, prop)) {
      continue;
    }

    const value = style[prop];

    if (PROPERTIES_TO_SWAP[prop]) {
      const newProp = flipProperty(prop);
      nextStyle[newProp] = value;
    } else if (PROPERTIES_SWAP_LEFT_RIGHT[prop]) {
      nextStyle[prop] = swapLeftRight(value);
    } else if (prop === 'textShadowOffset') {
      nextStyle[prop] = value;
      nextStyle[prop].width = additiveInverse(value.width);
    } else {
      nextStyle[prop] = style[prop];
    }
  }

  return nextStyle;
};

export default i18nStyle;
