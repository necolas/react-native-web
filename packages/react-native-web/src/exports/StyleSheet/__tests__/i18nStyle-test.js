/* eslint-env jasmine, jest */

import I18nManager from '../../I18nManager';
import i18nStyle from '../i18nStyle';

describe('StyleSheet/i18nStyle', () => {
  describe('isRTL is false', () => {
    beforeEach(() => {
      I18nManager.allowRTL(false);
    });

    afterEach(() => {
      I18nManager.allowRTL(true);
    });

    test('converts end/start properties', () => {
      const initial = {
        borderStartColor: 'red',
        start: 1,
        marginStart: 5,
        paddingEnd: 10
      };
      const expected = {
        borderLeftColor: 'red',
        left: 1,
        marginLeft: 5,
        paddingRight: 10
      };
      expect(i18nStyle(initial)).toEqual(expected);
    });

    test('converts end/start values', () => {
      const initial = {
        float: 'start',
        textAlign: 'end'
      };
      const expected = {
        float: 'left',
        textAlign: 'right'
      };
      expect(i18nStyle(initial)).toEqual(expected);
    });

    test('noop on left/right properties', () => {
      const initial = {
        paddingLeft: 0,
        left: 0,
        marginRight: 0,
        paddingRight: 10
      };
      expect(i18nStyle(initial)).toEqual(initial);
    });

    test('noop on left/right values', () => {
      const initial = {
        clear: 'left',
        float: 'left',
        textAlign: 'right',
        textShadowOffset: { width: '1rem', height: 10 }
      };
      expect(i18nStyle(initial)).toEqual(initial);
    });

    test('end/start properties take precedence', () => {
      const initial = {
        borderStartWidth: 10,
        borderLeftWidth: 0,
        end: 10,
        right: 0,
        marginStart: 10,
        marginLeft: 0
      };
      const expected = {
        borderLeftWidth: 10,
        marginLeft: 10,
        right: 10
      };
      expect(i18nStyle(initial)).toEqual(expected);
    });
  });

  describe('isRTL is true', () => {
    beforeEach(() => {
      I18nManager.forceRTL(true);
    });

    afterEach(() => {
      I18nManager.forceRTL(false);
    });

    describe('doLeftAndRightSwapInRTL is false', () => {
      beforeEach(() => {
        I18nManager.swapLeftAndRightInRTL(false);
      });

      afterEach(() => {
        I18nManager.swapLeftAndRightInRTL(true);
      });

      test('converts end/start properties', () => {
        const initial = {
          borderStartColor: 'red',
          start: 1,
          marginStart: 5,
          paddingEnd: 10
        };
        const expected = {
          borderRightColor: 'red',
          right: 1,
          marginRight: 5,
          paddingLeft: 10
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('converts end/start values', () => {
        const initial = {
          float: 'start',
          textAlign: 'end'
        };
        const expected = {
          float: 'right',
          textAlign: 'left'
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('noop on left/right properties', () => {
        const initial = {
          paddingLeft: 0,
          left: 0,
          marginRight: 0,
          paddingRight: 10
        };
        expect(i18nStyle(initial)).toEqual(initial);
      });

      test('noop on left/right values', () => {
        const initial = {
          clear: 'left',
          float: 'left',
          textAlign: 'right',
          textShadowOffset: { width: '1rem', height: 10 }
        };
        expect(i18nStyle(initial)).toEqual(initial);
      });

      test('end/start properties take precedence', () => {
        const style = {
          borderStartWidth: 10,
          borderRightWidth: 0,
          end: 10,
          left: 0,
          marginStart: 10,
          marginRight: 0
        };
        const expected = {
          borderRightWidth: 10,
          marginRight: 10,
          left: 10
        };
        expect(i18nStyle(style)).toEqual(expected);
      });
    });

    describe('doLeftAndRightSwapInRTL is true', () => {
      test('converts end/start properties', () => {
        const initial = {
          borderStartColor: 'red',
          start: 1,
          marginStart: 5,
          paddingEnd: 10
        };
        const expected = {
          borderRightColor: 'red',
          right: 1,
          marginRight: 5,
          paddingLeft: 10
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('converts end/start values', () => {
        const initial = {
          float: 'start',
          textAlign: 'end'
        };
        const expected = {
          float: 'right',
          textAlign: 'left'
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('converts left/right properties', () => {
        const initial = {
          borderLeftColor: 'red',
          left: 1,
          marginLeft: 5,
          paddingRight: 10
        };
        const expected = {
          borderRightColor: 'red',
          right: 1,
          marginRight: 5,
          paddingLeft: 10
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('converts left/right values', () => {
        const initial = {
          float: 'left',
          textAlign: 'right',
          textShadowOffset: { width: '1rem', height: 10 }
        };
        const expected = {
          float: 'right',
          textAlign: 'left',
          textShadowOffset: { width: '-1rem', height: 10 }
        };
        expect(i18nStyle(initial)).toEqual(expected);
      });

      test('end/start properties take precedence', () => {
        const style = {
          borderStartWidth: 10,
          borderLeftWidth: 0,
          end: 10,
          right: 0,
          marginStart: 10,
          marginLeft: 0
        };
        const expected = {
          borderRightWidth: 10,
          marginRight: 10,
          left: 10
        };
        expect(i18nStyle(style)).toEqual(expected);
      });
    });
  });
});
