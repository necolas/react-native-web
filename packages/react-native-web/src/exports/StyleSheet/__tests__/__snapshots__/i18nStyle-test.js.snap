// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apis/StyleSheet/i18nStyle LTR mode does not auto-flip 1`] = `
Object {
  "borderBottomLeftRadius": 20,
  "borderBottomRightRadius": "2rem",
  "borderLeftColor": "red",
  "borderLeftStyle": "solid",
  "borderLeftWidth": 5,
  "borderRightColor": "blue",
  "borderRightStyle": "dotted",
  "borderRightWidth": 6,
  "borderTopLeftRadius": 10,
  "borderTopRightRadius": "1rem",
  "left": 1,
  "marginLeft": 7,
  "marginRight": 8,
  "paddingLeft": 9,
  "paddingRight": 10,
  "right": 2,
  "textAlign": "left",
  "textShadowOffset": Object {
    "height": 10,
    "width": "1rem",
  },
}
`;

exports[`apis/StyleSheet/i18nStyle RTL mode does auto-flip 1`] = `
Object {
  "borderBottomLeftRadius": "2rem",
  "borderBottomRightRadius": 20,
  "borderLeftColor": "blue",
  "borderLeftStyle": "dotted",
  "borderLeftWidth": 6,
  "borderRightColor": "red",
  "borderRightStyle": "solid",
  "borderRightWidth": 5,
  "borderTopLeftRadius": "1rem",
  "borderTopRightRadius": 10,
  "left": 2,
  "marginLeft": 8,
  "marginRight": 7,
  "paddingLeft": 10,
  "paddingRight": 9,
  "right": 1,
  "textAlign": "right",
  "textShadowOffset": Object {
    "height": 10,
    "width": "-1rem",
  },
}
`;
